javascript/javascript3/week1/Giphy api.js

```diff
@@ -0,0 +1,58 @@
+"use strict";
+/* Steps to break the problem into smaller ones:
+1. create html,css and js file.
+2. create  an input and a button in the html file.
+3. checking the website and getting the url to fetch it in the js file
+4. check if it needs signing up to get the key for the url
+5. after fetching the url, creating a function where it displays a certain gif related to the searched word.
+6. styling the page in css.  */
+
+const key = "xqp6W3VtKgpQ3BlfimA85Eg0sl3LxMhW";
+
+const button = document
+  .getElementById("giphy-button")
+  .addEventListener("click", getGiphy);
+
+const gifInput = document.getElementById("giphy-input");
+const gifNumber = document.getElementById("giphy-number");
+
+function getGiphy() {
+  //   e.preventDefault();
+  let input = gifInput.value;
+  let number = gifNumber.value;
+  let url = `https://api.giphy.com/v1/gifs/search?api_key=${key}&q=${input}&limit=${number}&offset=0&rating=g&lang=en`;
+  console.log(url);
+  fetch(url)
+    .then((response) => response.json())
+    .then((content) => {
+      content.data.forEach((element) => {
+        // console.log(content.data);
+        // console.log(content.meta);
+        if (number === "") {
```

Comment:
I think you don't need this check here . You can use it before fetch if you don't want to proceed further without limit otherwise this api returns you default number of search results even you have not explicitly set the search limit .

Reply:
Thank you but this check here makes sure to specify a number or it will give you all the gifs with that name which is 50 or 80 ! to make sure it doesnt give you all the gifs , I came up with this idea.